pandas/core/indexes/multi.py

```diff
@@ -89,6 +89,9 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
             raise TypeError("Must pass both levels and labels")
         if len(levels) != len(labels):
             raise ValueError('Length of levels and labels must be the same.')
+        if names is not None and len(names) > 0 and len(levels) == 0:
+            levels = [[] for _ in names]
+            labels = [[] for _ in names]
```

Comment:
@jreback is this the PR you're thinking of: PR #16486? It looks like that change to `_shallow_copy_with_infer` could be refactored to simply ensuring metadata was propagated through to `_shallow_copy`.